### PR TITLE
Don't attempt to parse `max_connection_lifetime`

### LINKED ISSUE #152: Changing postgresql connection_url causes a crash
### Terraform Version
```
Terraform v0.11.7
+ provider.vault (unversioned) 
```
Using 107958019ce8060666d61e029c663d5d99deaab4
### Affected Resource(s)
Please list the resources as a list, for example:
- `vault_database_secret_backend_connection`

### Terraform Configuration Files
```hcl
provider "vault" {
  address = "http://localhost:8200"
}

resource "vault_mount" "postgres" {
  path = "database"
  type = "database"
}

resource "vault_database_secret_backend_connection" "postgres" {
  backend       = "${vault_mount.postgres.path}"
  name          = "postgres"
  allowed_roles = ["*"]

  postgresql {
    connection_url = "postgres://postgres:postgres@localhost:5432/postgres?sslmode=disable"
  }
}
```

### Debug Output
https://gist.github.com/Mongey/9aee3c5f9dbbb4adb81254942748bbaa

### Panic Output
https://gist.github.com/Mongey/9aee3c5f9dbbb4adb81254942748bbaa#file-crash-log

### Expected Behavior
Connection should have been updated

### Actual Behavior
💣 

### Steps to Reproduce

0. `docker run -it p 5432:5432 postgres`
1. `terraform apply`
2. Change the `connection_url `
3. terraform apply

### PR DESCRIPTION
**Pull request by Mongey**
Fixes #152

### CODE REVIEW

**Comment by catsby** [comment]
Looks good to me! Could I trouble you to update one of the acceptances tests to introduce the `max_connection_lifetime` to a configuration and verify this passes now? 

- https://github.com/terraform-providers/terraform-provider-vault/blob/107958019ce8060666d61e029c663d5d99deaab4/vault/resource_database_secret_backend_connection_test.go#L92

**Comment by myoung34** [comment]
Any updates to this? Its affecting our staging rebuild =(

**Comment by Mongey** [comment]
Looks like this is the same issues as described in #117 😬

File: vault/resource_database_secret_backend_connection_test.go [hunk 1]
```diff
@@ -24,7 +24,7 @@ func TestAccDatabaseSecretBackendConnection_import(t *testing.T) {
 		CheckDestroy: testAccDatabaseSecretBackendConnectionCheckDestroy,
 		Steps: []resource.TestStep{
 			{
-				Config: testAccDatabaseSecretBackendConnectionConfig_postgresql(name, backend, connURL),
+				Config: testAccDatabaseSecretBackendConnectionConfig_postgresql(name, backend, connURL, 0),
```
Comment (tyrannosaurus-becks) [inline vault/resource_database_secret_backend_connection_test.go]: I ran this one locally and got a test failure. Initially it was just about the connection url in plain text not matching a redacted one, so I commented that out (on line 35) and re-ran it and got this:
```
=== RUN   TestAccDatabaseSecretBackendConnection_import
--- FAIL: TestAccDatabaseSecretBackendConnection_import (0.03s)
    testing.go:434: Step 0 error: After applying this step, the plan was not empty:
        
        DIFF:
        
        UPDATE: vault_database_secret_backend_connection.test
          postgresql.0.connection_url: "*****://*****:*****@localhost:5432/*****" => "postgres://postgres:postgres@localhost:5432/postgres"
        
        STATE:
        
        vault_database_secret_backend_connection.test:
          ID = tf-test-db-3758954475378231597/config/db-7907426470007612753
          allowed_roles.# = 2
          allowed_roles.0 = dev
          allowed_roles.1 = prod
          backend = tf-test-db-3758954475378231597
          name = db-7907426470007612753
          postgresql.# = 1
          postgresql.0.connection_url = *****://*****:*****@localhost:5432/*****
          postgresql.0.max_connection_lifetime = 0
          postgresql.0.max_idle_connections = 0
          postgresql.0.max_open_connections = 2
          verify_connection = true
        
          Dependencies:
            vault_mount.db
        vault_mount.db:
          ID = tf-test-db-3758954475378231597
          path = tf-test-db-3758954475378231597
          type = database
```

Comment (coryflucas) [reply vault/resource_database_secret_backend_connection_test.go]: I believe https://github.com/terraform-providers/terraform-provider-vault/pull/217 should resolve this test issue as its also due to the redacted URL.

**Comment by tyrannosaurus-becks** [comment]
@Mongey should we still keep this open in light of #217 being merged?

**Comment by coryflucas** [comment]
@tyrannosaurus-becks I believe this resolves another issue.  #217 stopped Terraform from thinking it needed to apply changes to a `vault_database_secret_backend_connection` when nothing had actually changed, but I believe it still fails if you try and update a `vault_database_secret_backend_connection` resource in general.

**Comment by tyrannosaurus-becks** [comment]
Closing as the dupe PR ended getting merged first! Thank you though!

**Pull request closed without merging**